test/jdk/java/awt/font/LineBreakMeasurer/LineBreakWithTracking.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @key headful
+  @bug 8165943
+  @summary LineBreakMeasurer does not measure correctly if TextAttribute.TRACKING is set
+  @compile LineBreakWithTracking.java
+  @run main/manual LineBreakWithTracking
+*/
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.font.FontRenderContext;
+import java.awt.font.LineBreakMeasurer;
+import java.awt.font.TextAttribute;
+import java.awt.font.TextLayout;
+import java.text.AttributedString;
+import java.util.Hashtable;
+
+class LineBreakPanel extends JPanel implements ActionListener {
+
+  private float textTracking = 0.0f;
+  private static String fontName = "Bitstream Cyberbit";
+  private static String text = "This is a long line of text that should be broken across multiple lines. "
+          + "Please set the different tracking values to test via menu! This test should pass if "
+          + "these lines are broken to fit the width, and fail otherwise.  It should "
+          + "also format the hebrew (\u05d0\u05d1\u05d2 \u05d3\u05d4\u05d5) and arabic "
+          + "(\u0627\u0628\u062a\u062c \u062e\u0644\u0627\u062e) and CJK "
+          + "(\u4e00\u4e01\u4e02\uac00\uac01\uc4fa\u67b1\u67b2\u67b3\u67b4\u67b5\u67b6\u67b7"
+          + "\u67b8\u67b9) text correctly.";
+
+  private LineBreakMeasurer lineMeasurer;
+
+  public void actionPerformed(ActionEvent e) {
+    textTracking = (float)((JRadioButtonMenuItem)e.getSource()).getClientProperty( "tracking" );
+    lineMeasurer = null;
+    invalidate();
+    repaint();
+  }
+
+  public void paintComponent(Graphics g) {
+    super.paintComponent(g);
+    setBackground(Color.white);
+
+    Graphics2D g2d = (Graphics2D)g;
+
+    if (lineMeasurer == null) {
+      Float regular = new Float(16.0);
+      Float big = new Float(24.0);
+
```

Comment:
LineBreakWithTracking.java:74: warning: [removal] Float(double) in Float has been deprecated and marked for removal
      Float regular = new Float(16.0);
                      ^
LineBreakWithTracking.java:75: warning: [removal] Float(double) in Float has been deprecated and marked for removal
      Float big = new Float(24.0);

Use Float.valueOf(float) instead

Reply:
fixed